activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -28,9 +28,16 @@ module Associations
     # is computed directly through SQL and does not trigger by itself the
     # instantiation of the actual post records.
     class CollectionProxy < Relation
+      @@_extensions = {}
```

Comment:
Maybe this would create a race condition when defining two extensions to the same association? @matthewd WDYT?